src/coreclr/src/jit/simdashwintrinsic.cpp

```diff
@@ -0,0 +1,1090 @@
+//    methodName         -- The name of the method associated with the SimdIntrinsic to lookup
+//    enclosingClassName -- The name of the enclosing class
+//    sizeOfVectorT      -- The size of Vector<T> in bytes
+//
+// Return Value:
+//    The NamedIntrinsic associated with methodName and classId
+NamedIntrinsic SimdAsHWIntrinsicInfo::lookupId(CORINFO_SIG_INFO* sig,
+                                               const char*       className,
+                                               const char*       methodName,
+                                               const char*       enclosingClassName,
+                                               int               sizeOfVectorT)
+{
+    SimdAsHWIntrinsicClassId classId = lookupClassId(className, enclosingClassName, sizeOfVectorT);
+
+    if (classId == SimdAsHWIntrinsicClassId::Unknown)
+    {
+        return NI_Illegal;
+    }
+
+    for (int i = 0; i < (NI_SIMD_AS_HWINTRINSIC_END - NI_SIMD_AS_HWINTRINSIC_START - 1); i++)
+    {
+        const SimdAsHWIntrinsicInfo& intrinsicInfo = simdAsHWIntrinsicInfoArray[i];
+
+        if (classId != intrinsicInfo.classId)
+        {
+            continue;
+        }
+
+        if (sig->numArgs != static_cast<unsigned>(intrinsicInfo.numArgs))
+        {
+            continue;
+        }
+
+        if (sig->hasThis() != SimdAsHWIntrinsicInfo::IsInstanceMethod(intrinsicInfo.id))
+        {
+            continue;
+        }
+
+        if (strcmp(methodName, intrinsicInfo.name) != 0)
+        {
+            continue;
+        }
+
+        return intrinsicInfo.id;
+    }
+
+    return NI_Illegal;
+}
+
+//------------------------------------------------------------------------
+// lookupClassId: Gets the SimdAsHWIntrinsicClassId for a given class name and enclsoing class name
+//
+// Arguments:
+//    className          -- The name of the class associated with the SimdAsHWIntrinsicClassId to lookup
+//    enclosingClassName -- The name of the enclosing class
+//    sizeOfVectorT      -- The size of Vector<T> in bytes
+//
+// Return Value:
+//    The SimdAsHWIntrinsicClassId associated with className and enclosingClassName
+SimdAsHWIntrinsicClassId SimdAsHWIntrinsicInfo::lookupClassId(const char* className,
+                                                              const char* enclosingClassName,
+                                                              int         sizeOfVectorT)
+{
+    assert(className != nullptr);
+
+    if ((enclosingClassName != nullptr) || (className[0] != 'V'))
+    {
+        return SimdAsHWIntrinsicClassId::Unknown;
+    }
+    if (strcmp(className, "Vector2") == 0)
+    {
+        return SimdAsHWIntrinsicClassId::Vector2;
+    }
+    if (strcmp(className, "Vector3") == 0)
+    {
+        return SimdAsHWIntrinsicClassId::Vector3;
+    }
+    if (strcmp(className, "Vector4") == 0)
+    {
+        return SimdAsHWIntrinsicClassId::Vector4;
+    }
+    if (strcmp(className, "Vector`1") == 0)
```

Comment:
This should also check against `"Vector"` to catch the corresponding methods in the static `Vector` class, will fix.